### PR TITLE
Wallet: dApps Tab

### LINKED ISSUE #17584: Wallet: dApps tab
Implement the following in dApps tab inside an account

- [x] dApps list
- [x] Options menu

Designs: https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2221-465823&mode=design&t=HaGGbeEtGuYsaw24-0

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/17584

This PR implements dApps tab UI and its options menu UI

[Designs](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2221-465824&mode=design&t=mSZhHoiV4wjvcXYa-4)

Demo:

https://github.com/status-im/status-mobile/assets/29354102/a396eb6c-61fc-4b4b-86ad-1a8b80164ca5

### CODE REVIEW

File: src/quo2/components/settings/category/reorder/view.cljs
Comment (smohamedjavid) [inline src/quo2/components/settings/category/reorder/view.cljs:1]: @OmarBasem - Any reason to remove the blur from the component?

**Review by smohamedjavid — APPROVED** [review]
LGTM 🚀

File: src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+(ns status-im2.contexts.wallet.account.tabs.dapps.style
+  (:require
+    [quo2.foundations.colors :as colors]
+    [status-im2.contexts.wallet.account.style :as common-style]))
+
+(def empty-container-style common-style/empty-container-style)
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs]: Imo we should avoid this pattern, it's not expected to have co dependencies between styles, instead let's encapsulate styles within components and use this approach to share styling across many places. 
e,g make a component `(defn empty-container [{} children] ...  [children])` 👍

File: src/status_im2/contexts/wallet/account/tabs/dapps/view.cljs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+(ns status-im2.contexts.wallet.account.tabs.dapps.view
+  (:require [quo2.core :as quo]
+            [quo2.theme :as quo.theme]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]
+            [status-im2.contexts.wallet.account.tabs.dapps.style :as style]
+            [status-im2.contexts.wallet.common.temp :as temp]
+            [utils.re-frame :as rf]))
+
+(defn dapp-options
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/account/tabs/dapps/view.cljs:11]: `defn-` ?

File: src/quo2/components/settings/category/reorder/view.cljs
Comment (OmarBasem) [reply src/quo2/components/settings/category/reorder/view.cljs:1]: This was for preview. It should be coming from the preview screen

File: src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs (hunk 1 above)
Comment (OmarBasem) [reply src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs]: @J-Son89 fixed it

**Review by Francesca-G — APPROVED** [review]
Here's the [design review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=6881%3A2858&mode=design&t=hURKF0jiZQHF32DZ-1) :)

**Review by vkjr — APPROVED** [review]

File: src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs (hunk 1 above)
Comment (J-Son89) [reply src/status_im2/contexts/wallet/account/tabs/dapps/style.cljs]: thanks!

File: src/quo2/components/settings/category/reorder/view.cljs [hunk 3]
```diff
@@ -20,8 +19,6 @@
   [{:keys [label data blur? theme]}]
   (reagent/with-let [atom-data (reagent/atom data)]
     [rn/view {:style style/container}
-     (when blur?
```
Comment (J-Son89) [inline src/quo2/components/settings/category/reorder/view.cljs:23]: sorry if this was already answered, but why are we deleting this blur here? t's wrong or something? 🤔

**Review by J-Son89 — APPROVED** [review]
looks good to me. 👍

File: src/status_im2/contexts/wallet/common/empty_tab/view.cljs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+(ns status-im2.contexts.wallet.common.empty-tab.view
+  (:require [quo2.core :as quo]
+            [status-im2.contexts.wallet.common.empty-tab.style :as style]))
+
+(defn view
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/common/empty_tab/view.cljs:5]: 🙌

File: src/status_im2/contexts/wallet/common/empty_tab/view.cljs [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  [props]
+  [quo/empty-state
+   (merge props {:container-style style/empty-container-style})])
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/common/empty_tab/view.cljs]: nit assoc 🤨

File: src/quo2/components/settings/category/reorder/view.cljs (hunk 3 above)
Comment (OmarBasem) [reply src/quo2/components/settings/category/reorder/view.cljs:23]: @J-Son89 yes it shouldn't be here. This was added for preview, and we now have blur automatically added in our `preview-container` when needed

Comment (J-Son89) [reply src/quo2/components/settings/category/reorder/view.cljs:23]: Nice 👌

**Pull request merged**